hubspot/crm/tool.gpt

```diff
@@ -0,0 +1,54 @@
+Name: HubSpot CRM
+Description: Tools for working with HubSpot CRM
+Metadata: bundle: true
+Share Context: HubSpot CRM Context
+Share Tools: ./associations.gpt, ./companies.gpt, ./contacts.gpt, ./deals.gpt, lists.gpt, owners.gpt, ./properties.gpt, ./tickets.gpt
+Share Tools: Get Me, Get Unix Timestamp, github.com/gptscript-ai/context/current-time
+
+---
+Name: HubSpot CRM Context
+Type: Context
+#!sys.echo
+
+## Instructions for using HubSpot CRM tools
+
+- Always call the 'HubSpot CRM Get Properties' tool to get schemas for a resource before creating/modifying it.
+- Do not return IDs to the user unless they specifically ask - they are generally not useful.
+- Remember the ID to use in the future to make additional edits or lookups.
+- Before making destructive changes, confirm with the user.
+- When creating or updating a resource, use the `Get Property By Name` tool if needed to get information about fields and the values that are valid for them.
+- For operations that require an `Id`, first use a tool to search if the user does not provide the Id.
+- Use the current user's ownerId when they ask for resources assigned to them.
+- When searching with strings, always use `CONTAINS_TOKEN` as the filter operator.
+- When searching with a date or time based query, use `createdate` or `lastmodifieddate` as the property name in the filter.
+- Use current-time when being asked to perform time-based operations.
+- When relevant, set `hs_timestamp` to the output of the `getUnixTimestamp` tool.
+- When looking for type of resource (for example, a ticket or a deal) associated with a specific object (for example, a contact or a company), use the `Get Association Type From Object ID` tool before using the search tools.
+- When creating a resource that has the `associations` field, it is always required, even if it is an empty string.
+- Ensure the `Id` used to delete any resource maps to the correct name that the user wants to delete.
+- Always use the `Get AssociationType ID` tool to look up associationTypeIds between two objects. Do not guess.
+
+## End of instructions for using HubSpot CRM tools
+
+---
+name: Get Me
+Type: context
+description: Gets details about the currently authenticated HubSpot user
+Credential: ./credential
+
+#!/usr/bin/env python3 ${GPTSCRIPT_TOOL_DIR}/getMe.py
+
+---
+name: Get Unix Timestamp
+description: Get the current Unix timestamp in milliseconds
+
+#!/usr/bin/env python3
+
+import time
+
+unix_timestamp_ms = round(time.time() * 1000)
+print(f"The current unix timestamp in milliseconds is {unix_timestamp_ms}")
```

Comment:
Not sure if this tool was ported over from somewhere else, but are we going to run into timezone issues with this? Should this be in the `Time` tool and use the timezone stuff added there?

Reply:
It was ported over from the original version of these tools, but a unix timestamp doesnt have a timezone anyway so it shouldnt cause any problems. arguably it could be part of the time tool, but its also somewhat specific to hubspot, because they want the unix timestamp in milliseconds, where normally an unix timestamp is in seconds